### PR TITLE
Upgrade base Python version to 3.10

### PR DESCRIPTION
**Pull request by ulgens**

### CODE REVIEW

**Review by cclauss — APPROVED** [review]

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -11,10 +11,10 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python 3.9
+    - name: Set up Python 3.10
       uses: actions/setup-python@v2
       with:
-        python-version: 3.9
+        python-version: "3.10"
```
Comment (cclauss) [inline .github/workflows/deploy.yml:17]:
```suggestion
        python-version: 3.x
```
Force the upgrade to the current Python as soon as it changes.

**Comment by ulgens** [comment]
@cclauss Can i get another review please? Thank you.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -11,10 +11,10 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python 3.9
+    - name: Set up Python
       uses: actions/setup-python@v2
```
Comment (cclauss) [inline .github/workflows/deploy.yml:15]: OPTIONAL:
```suggestion
      uses: actions/setup-python@v4
```
https://github.com/actions/setup-python/releases

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -8,10 +8,10 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python 3.9
+    - name: Set up Python 3.10
       uses: actions/setup-python@v2
```
Comment (cclauss) [inline .github/workflows/lint.yml:12]: OPTIONAL:
```suggestion
      uses: actions/setup-python@v4
```

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -11,10 +11,10 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
```
Comment (cclauss) [inline .github/workflows/deploy.yml:13]: OPTIONAL:
```suggestion
    - uses: actions/checkout@v3
```
https://github.com/actions/checkout/releases

**Review by cclauss — DISMISSED** [review]
LGTM

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (ulgens) [reply .github/workflows/deploy.yml:15]: I think those updates are better suited for a different PR.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (erikwrede) [reply .github/workflows/deploy.yml]: I don't agree on this, we should keep everything on the same version since some steps have to be upgraded manually.

Comment (ulgens) [reply .github/workflows/deploy.yml]: It's not a strong opinion, but I'm on the more explicit part with this.

Comment (ulgens) [reply .github/workflows/deploy.yml:17]: @erikwrede Rollbacked the 3.x change, should be okay now.

**Review by erikwrede — APPROVED** [review]

**Pull request merged**